### PR TITLE
[Tooling] add playwright tests into azure pipeline

### LINKED ISSUE #7999: 🛠️ Run an automated test in Azure
## 🛠️ Tooling

We recently had a few near misses with issues found in UAT in Azure before deployment.  There was the auth issue where the first request after logging in failed.  And then the database upgrade with the misconfigured extension.  I was thinking that it would be handy if we could run a batch of tests like ~Cypress~ Playwright against the dev or uat instance in Azure.

If the dev vertical was open to the Internet I wonder if it would be possible to open Cypress on a dev's machine and point it at the Azure instance?

## ✅ Acceptance Criteria

- [ ] How do we manually  run a batch of automated tests against an instance running in Azure Dev 

## Timebox 
3 days

### PR DESCRIPTION
**Pull request by brindasasi**
🤖 Resolves #7999 

## 👋 Introduction

This PR integrates playwright tests into Azure pipeline so that the tests can be triggered against the dev verticals.
Currently it runs against dev environment.

## 🕵️ Details

Below is the Dev Pipeline that got set up to run playwright tests
image

Basically the pipeline got split into 2 stages as Build and Test stage.
**Build stage** 
  Builds the latest main and creates the artifact 
**Test Stage** 
  This stage downloads the artifact that created on the prev stage and feeds off to a self hosted VM.
**Self Hosted VM** 
   This VM is set up to run playwright tests with all the required dependencies and white listed from the VPN sites like the dev environment.
   The tests results are stored in VM but now got uploaded in Azure Storage container for easy access 

**Storage Container Path** 
https://portal.azure.com/#@056gc.onmicrosoft.com/resource/subscriptions/e45226c5-f9e9-4979-9016-96ba27b156cb/resourceGroups/rg_tbs_talent_2_PaaS/providers/Microsoft.Storage/storageAccounts/sttalclddevcacstdl/containersList

This PR just set up infrastructure for E2E tests running against dev verticals.

**Tests needs to be modified to run against GCKey logins in a diff issue.**

## 🧪 Testing

**Need Azure access to test this

1.  Log into dev.azure.com/tbs-sct with your PC account
2. Run the dev pipeline 
3. Verify the results in the command line 
4. Verify the screenshots in the storage container with steps below 

- RDP into either of the VM T1_Dev_Management or T1_Management using your PS account 
- Open Edge and login to  your PC account 
- click on the link  https://portal.azure.com/#@056gc.onmicrosoft.com/resource/subscriptions/e45226c5-f9e9-4979-9016-96ba27b156cb/resourceGroups/rg_tbs_talent_2_PaaS/providers/Microsoft.Storage/storageAccounts/sttalclddevcacstdl/containersList

## 📸 Screenshot

image

image (2)

image (3)

### CODE REVIEW

File: infrastructure/.npmrc
Comment (petertgiles) [inline infrastructure/.npmrc:1]: We're not using npm anymore.

File: infrastructure/bin/deploy.sh
Comment (petertgiles) [inline infrastructure/bin/deploy.sh:77]: This command doesn't exist manymore since the "reduce artifact size" project.

File: infrastructure/azure-pipelines-dev.yml
Comment (petertgiles) [inline infrastructure/azure-pipelines-dev.yml:25]: Did you mean to remove this comment?

Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:25]: hmm.. no .. will put it back

**Comment by brindasasi** [comment]
oof.. It had to go through another cycle of build issues with npm->pnpm
hope we don't switch to another PM before this PR gets merged in 😁
Daniel will block me!

File: infrastructure/.npmrc
Comment (brindasasi) [inline infrastructure/.npmrc:1]: btw, we still need this.. as we need npm to  install the pnpm asfaik

File: apps/playwright/.env.example
Comment (brindasasi) [inline apps/playwright/.env.example:6]: this is just my idea .. not using this in this PR .. separated fixing tests into another one as I'm focused to get it running in Azure in this PR. Open to ideas on storing the diff credentials.

File: apps/playwright/playwright.config.ts
Comment (petertgiles) [inline apps/playwright/playwright.config.ts:31]: Not sure this should be hardcoded into this file?

File: infrastructure/azure-pipelines-dev.yml
Comment (petertgiles) [inline infrastructure/azure-pipelines-dev.yml:103]: Does pipeline actually run cypress?

**Review by petertgiles — COMMENTED** [review]
Looking cool!  Is this branch ready for review?  It looks like the pipeline in Azure isn't completing successfully yet.

File: infrastructure/azure-pipelines-dev.yml
Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:103]: no .. I think we are good to remove it

Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:103]: backing out this fails on the dependency installation on the pipeline.. looks like we need it

File: infrastructure/azure-pipelines-dev.yml
Comment (petertgiles) [inline infrastructure/azure-pipelines-dev.yml:105]: Why does it install dependencies and rebuild the app a second time in this stage?  Isn't that the purpose of the first stage?

**Review by petertgiles — COMMENTED** [review]
Very neat.  I just read through and the code and launched the pipeline again.  I had a few more questions.

**Review by esizer — APPROVED** [review]
Can confirm playwright stuff looks great :+1: I'd just wait on @petertgiles approval though, I just looked at the local playwright config :sweat_smile:

File: infrastructure/azure-pipelines-dev.yml
Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:105]: Reason is that we had to install the deps on the agent that is running the tests. There was some security issue with the Talentcloud project that forced the deps downloaded from the agent instead of from the feed. 
Same steps with another test pipeline works without having to install the deps but not with our TalentCloud project

Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:105]: I will try get the convo from Daniel  related to that.

Comment (petertgiles) [reply infrastructure/azure-pipelines-dev.yml:105]: Can we drop the first stage entirely then?  Is there any point to it if we're just going to rebuild it again?

Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:105]: image (2)

Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:105]: image (3)

File: apps/playwright/.env.example
Comment (brindasasi) [reply apps/playwright/.env.example:5]: I'm giving it in here https://github.com/GCTC-NTGC/gc-digital-talent/blob/69a2b2b73f1318e0cdfc154d0a78a9d9341b7c25/infrastructure/azure-pipelines-dev.yml#L118

File: infrastructure/azure-pipelines-dev.yml
Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:105]: > Can we drop the first stage entirely then? Is there any point to it if we're just going to rebuild it again?

would check with Daniel on that.

Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:105]: yeah Daniel also feels that its redundant at this point to have that stage. Will try disconnect it first if it passes then it should be completely good to go. Thanks for catching that!

Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:100]: hmm doesn't work .. complaining it misses graphql dep.

Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml]: removed build stage in [53d7f44](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9652/commits/53d7f44b00cc93c53eea97403cd94d97ef68195f) !

File: infrastructure/azure-pipelines-dev.yml
Comment (petertgiles) [inline infrastructure/azure-pipelines-dev.yml:17]: Does this container automatically clean up old results?

File: infrastructure/azure-pipelines-dev.yml
Comment (petertgiles) [inline infrastructure/azure-pipelines-dev.yml:23]: It seems like this is not long enough.  It's timing out for me, though it completes in around 25 minutes in GH.

**Review by petertgiles — APPROVED** [review]
I'm not sure how this is going to work out.  :sweat_smile: If you want to merge this pipeline as a work-in-progress it's not going to affect anything else, though.

File: infrastructure/azure-pipelines-dev.yml
Comment (brindasasi) [reply infrastructure/azure-pipelines-dev.yml:17]: we have set a lifecycle rule on the container level to delete older results. Currently it is more than 30 days old.

**Comment by brindasasi** [comment]
> I'm not sure how this is going to work out. 😅 If you want to merge this pipeline as a work-in-progress it's not going to affect anything else, though.

I know its a new horizon but I'm confident on it.  😺

**Pull request merged**
